lnwallet/channel.go

```diff
@@ -5996,16 +5996,49 @@ func (lc *LightningChannel) availableCommitmentBalance(view *htlcView) (
 	)
 	feePerKw := filteredView.feePerKw
 
+	// Set the dustlimit based on who's commitment we are looking at.
+	dustlimit := lnwire.NewMSatFromSatoshis(
+		lc.channelState.LocalChanCfg.DustLimit,
+	)
+
 	// Given the commitment weight, find the commitment fee in case of no
 	// added HTLC ouput.
 	baseCommitFee := lnwire.NewMSatFromSatoshis(
 		feePerKw.FeeForWeight(commitWeight),
 	)
 
+	// Also calculate the commitment fee in the case where we would add
+	// another HTLC large enough to create a new output on the commitment.
+	htlcCommitFee := lnwire.NewMSatFromSatoshis(
+		feePerKw.FeeForWeight(commitWeight + input.HTLCWeight),
+	)
+
+	// For an extra HTLC fee to be paid on our commitment, the HTLC must be
+	// large enough to make a non-dust HTLC timeout transaction.
+	htlcFee := lnwire.NewMSatFromSatoshis(
+		htlcTimeoutFee(feePerKw),
```

Comment:
When we compute for the remote party's commitment, we need to use the `htlcSuccessFee` here. I think we can actually just use `htlcIsDust` directly which accounts for the relative change in fee depending on the commitment we're factoring in.

Reply:
Changes in later commit. 

Will check if I can use `htlcIsDust`

Reply:
Looks like we cannot use `htlcIsDust` since it expects the `htlcAmount` as an argument, while we want the reverse: get the minimum sized HTLC that is non-dust.